onnxruntime/core/framework/graph_partitioner.cc

```diff
@@ -570,16 +593,27 @@ static Status PartitionOrtFormatModelImpl(const PartitionParams& partition_param
     }
   }
 
-  const std::string& type = current_ep.Type();
   std::vector<std::unique_ptr<ComputeCapability>> capabilities;
+
 #if !defined(ORT_MINIMAL_BUILD) || defined(ORT_EXTENDED_MINIMAL_BUILD)
-  TransformLayoutFunction transform_layout_function = partition_params.transform_layout_function;
-#else
-  TransformLayoutFunction transform_layout_function{};
-#endif
-  ORT_RETURN_IF_ERROR(GetCapabilityForEP(graph, kernel_registry_mgr, current_ep,
-                                         GraphPartitioner::Mode::kOrtFormatLoad, capabilities,
-                                         transform_layout_function));
+  const auto get_capability_params = GetCapabilityForEPParams{
+      std::ref(graph),
+      std::cref(kernel_registry_mgr),
+      std::ref(current_ep),
+      std::ref(capabilities),
+      GraphPartitioner::Mode::kOrtFormatLoad,
+      partition_params.transform_layout_function,
+  };
```

Comment:
Can we put the ifdef around these two params to reduce duplication?

Reply:
clang-format didn't like the ifdef there so I disabled it for those lines